test/e2e/aks_public_ip_prefix.go

```diff
@@ -89,10 +89,12 @@ func AKSPublicIPPrefixSpec(ctx context.Context, inputGetter func() AKSPublicIPPr
 			Namespace: input.Cluster.Namespace,
 		},
 		Spec: infrav1.AzureManagedMachinePoolSpec{
-			Mode:                 "User",
-			SKU:                  os.Getenv("AZURE_NODE_MACHINE_TYPE"),
-			EnableNodePublicIP:   ptr.To(true),
-			NodePublicIPPrefixID: ptr.To("/subscriptions/" + subscriptionID + "/resourceGroups/" + resourceGroupName + "/providers/Microsoft.Network/publicipprefixes/" + *publicIPPrefix.Name),
+			AzureManagedMachinePoolClassSpec: infrav1.AzureManagedMachinePoolClassSpec{
+				Mode:                 "User",
+				SKU:                  os.Getenv("AZURE_NODE_MACHINE_TYPE"),
+				EnableNodePublicIP:   ptr.To(true),
+				NodePublicIPPrefixID: ptr.To("/subscriptions/" + subscriptionID + "/resourceGroups/" + resourceGroupName + "/providers/Microsoft.Network/publicipprefixes/" + *publicIPPrefix.Name),
```

Comment:
Is there a function to construct this ID, i.e. maybe in defaults.go?

Reply:
I created a function for it! Nice catch